pkg/querier/queryrange/results_cache.go

```diff
@@ -184,7 +189,7 @@ func (s resultsCache) Do(ctx context.Context, r Request) (Response, error) {
 	}
 
 	if err == nil && len(extents) > 0 {
-		extents, err := s.filterRecentExtents(r, extents)
+		extents, err := s.filterRecentExtents(ctx, r, extents)
```

Comment:
Can you pass the duration instead. ? to avoid calling this twice `s.limits.MaxCacheFreshness(userID)` ?

Reply:
Sure, fixing this.